.github/actions/setup-nix/action.yml

```diff
@@ -20,46 +20,73 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Nix install mode
+      shell: bash
+      run: |
+        if [[ ${{ runner.os }} != 'Linux' || $USER == 'root' ]]; then
```

Comment:
I think the logic here is not as described in the commit message. The commit message says a Linux root would have single-user mode, but this gives it multi-user mode.